packages/mui-lab/src/TreeItem/TreeItem.js

```diff
@@ -272,7 +272,15 @@ const TreeItem = React.forwardRef(function TreeItem(inProps, ref) {
   function handleFocus(event) {
     // DOM focus stays on the tree which manages focus with aria-activedescendant
     if (event.target === event.currentTarget) {
-      ownerDocument(event.target).getElementById(treeId).focus({ preventScroll: true });
+      let rootElement;
+
+      if (typeof event.target.getRootNode === 'function') {
```

Comment:
AFAIK, [`getRootNode`](https://developer.mozilla.org/en-US/docs/Web/API/Node/getRootNode) is always a function regardless of whether it is outside or inside of shadow DOM.

Reply:
The element returned by getRootNode() outside a shadow root is the document, if the element is inside a shadow root it will be the shadow root.
The typeof check is for browser version without support for shadow roots.

Reply:
Okay, makes sense.